### PR TITLE
FINDO-8413 support PHP 7.4

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

**Comment by TheKeymaster** [comment]
@howard you have no idea how frustrating it is to work with these tests.

1. They are old.
1. They have tests that only execute on travis.
1. THESE TESTS ARE A HELL TO DEBUG.
1. Nice evening. It is 20:18 and I stop working now, I have enough of these tests. Hopefully I won't get nightmares of them..

File: HTTP/Request2/SocketWrapper.php [hunk 1]
```diff
@@ -85,19 +85,34 @@ public function __construct($address, $timeout, array $contextOptions = array())
...
-                             . 'ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:'
-                             . 'ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:'
-                             . 'ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:'
-                             . 'DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:'
-                             . 'DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:'
-                             . 'ECDHE-RSA-DES-CBC3-SHA:ECDHE-ECDSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:'
-                             . 'AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:'
-                             . 'AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:'
-                             . '!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA'
+                'ciphers' => 'ECDHE-ECDSA-CHACHA20-POLY1305:'
```
Comment (howard) [inline HTTP/Request2/SocketWrapper.php:88]: Are there any insecure cipher suites in this list?

**Review by howard — COMMENTED** [review]
Nice work! Please check if there are any insecure cipher suites still allowed, otherwise this looks good.

A PR of mine on the upstream `pear/HTTP_Request2` was accepted a while ago, so *someone* is doing something. Should we also submit this there?

File: HTTP/Request2/SocketWrapper.php (hunk 1 above)
Comment (TheKeymaster) [reply HTTP/Request2/SocketWrapper.php:88]: @howard I cherry-picked this change from here: https://github.com/findologic/HTTP_Request2/pull/3/commits/96afdbe38464cf846e9d9af5001e4b06e8e44d17. Actually they gave us some changes this time :sweat_smile:.

Comment (TheKeymaster) [reply HTTP/Request2/SocketWrapper.php:88]: Additionally there is a test that explicitly checks the cipher suites.

File: composer.json [hunk 2]
```diff
@@ -1,42 +1,53 @@
...
-    },
-    "include-path": [
-        "./"
-    ],
-    "extra": {
-        "branch-alias": {
-            "dev-trunk": "2.2-dev"
-        }
+  "name": "findologic/http_request2",
+  "description": "Provides an easy way to perform HTTP requests.",
```
Comment (wladi0097) [inline composer.json:3]: `Provides an easy way to perform HTTP requests.` -> `Provides an outdated way to perform HTTP requests.` :DD

**Review by wladi0097 — COMMENTED** [review]
lgtm

**Review by howard — APPROVED** [review]

**Pull request merged**